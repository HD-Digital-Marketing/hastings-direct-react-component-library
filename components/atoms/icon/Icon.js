import React from 'react'
import SVG from 'react-inlinesvg'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './icon.scss'

const Icon = props => {

    const {
        isRound = false,
        svgImage,
        svgWidth,
        svgHeight,
        svgFill,
        svgBackgroundColor,
        svgBorderColor,
        className = '',
        onClick
    } = props

    const svgParentInlineStyle = {
        backgroundColor: svgBackgroundColor,
        borderColor: svgBorderColor,
        borderWidth: svgWidth/100*10,
        width: svgBackgroundColor || svgBorderColor ? svgWidth + svgWidth : svgWidth,
        height: svgBackgroundColor || svgBorderColor ? svgHeight + svgHeight : svgHeight
    }

    const svgInlineStyle = {
        fill: svgFill
    }

    const iconClass = classNames('a-icon', className, {
        'a-icon--rounded': isRound
    })

    return(
        <span className={iconClass} style={svgParentInlineStyle} {...{onClick}}>
            <SVG src={svgImage} width={svgWidth} height={svgHeight} style={svgInlineStyle} />
        </span>
    )
}

Icon.propTypes = {
    isRound: PropTypes.bool,
    svgImage: PropTypes.string,
    svgWidth: PropTypes.number,
    svgHeight: PropTypes.number,
    svgFill: PropTypes.string,
    svgBackgroundColor: PropTypes.string,
    svgBorderColor: PropTypes.string,
    className: PropTypes.string
}

export default Icon
