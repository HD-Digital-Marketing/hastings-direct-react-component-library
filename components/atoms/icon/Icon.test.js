import React from 'react'
import renderer from 'react-test-renderer'
import Icon from './Icon'
import carIcon from "../../../src/img/icons/car.svg";

describe('Check the icon renders', () => {
    test('it should match the snapshot', () => {

        const tree = renderer
            .create(<Icon svgImage={carIcon} svgWidth={30} svgHeight={30} svgFill="white" svgBackgroundColor="#0085ff" isRound={true}/>)
            .toJSON()
        expect(tree).toMatchSnapshot()

    })
})