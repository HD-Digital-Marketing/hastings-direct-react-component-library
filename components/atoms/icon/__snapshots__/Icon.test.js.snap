// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the icon renders it should match the snapshot 1`] = `
<span
  className="a-icon a-icon--rounded"
  style={
    Object {
      "backgroundColor": "#0085ff",
      "borderColor": undefined,
      "height": 39,
      "width": 39,
    }
  }
/>
`;
